pkg/doctype/doctype.go

```diff
@@ -0,0 +1,27 @@
+package doctype
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+var doctypeLookupRe = regexp.MustCompile(`(?ms)(<\?.*?\?>)?\s*<!DOCTYPE\s+([a-zA-Z_][a-zA-Z0-9]*(?: \s+[a-zA-Z_][a-zA-Z0-9]*\s+"[^"]*")?)[^>]*>`)
+
+// MatchString check if the doctype matches a regular expression (if present).
+func MatchString(text string, pattern string) (bool, error) {
```

Comment:
As it is an exported function, can we have a test for this one as well?

Reply:
Forgot to add them.